addons/website_customer/controllers/main.py

```diff
@@ -85,15 +85,15 @@ def customers(self, country=None, industry=None, page=0, **post):
         countries = Partner.sudo().read_group(domain, ["id", "country_id"], groupby="country_id", orderby="country_id")
         country_count = Partner.sudo().search_count(domain)
 
+        fallback = False
         if country:
-            domain += [('country_id', '=', country.id)]
-            if country.id not in (x['country_id'][0] for x in countries if x['country_id']):
-                if country.exists():
-                    countries.append({
-                        'country_id_count': 0,
-                        'country_id': (country.id, country.name)
-                    })
-                    countries.sort(key=lambda d: (d['country_id'] or (0, ""))[1])
+            if country_count > 0 and country.exists() and country.id not in (x['country_id'][0] for x in countries if x['country_id']):
```

Comment:
So if country doesn't exist, we don't fall back? Seems odd to me

Reply:
The fallback is just about the following: If the user is searching for a customer in a specific country and there is no result in this country but there is at least one result in another country, we fall back on all countries.

So yes, if country doesn't exists, we don't fall back

Reply:
But if the country doesn't exist, won't it necessarily mean no result can be found for the country?
In that case wouldn't it make sense to fall back immediately too?

As it stands, we'll fall back whenever there's no result for the specific country, unless the country happens to be a trash value. Seems odd. Unless I'm misunderstanding the 'exists' flow

Reply:
In fact I think the previous line `if country.exists():` was already wrong.. As we go through the model converter I don't see how we can end up in a situation where country is not None and country doesn't exists ? If the url has a trash country value, the dispatcher respond 404 directly so yes you are right, I will remove the exists condition :+1: